homework/dima_kachenya/bookRepositoriy/src/com/kachenya/book/BookRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.kachenya.book;
+
+import com.kachenya.customExeprion.ItemNotFoundExeption;
+import com.kachenya.repository.Repository;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+public class BookRepository<E extends Book> implements Repository {
+    HashMap<String, Book> myCollection = new HashMap();
+
+    @Override
+    public List findAll() {
+        List<Book> myList = new LinkedList<>();
+        for (Book item :
+                myCollection.values()) {
+            myList.add(item);
+        }
+
+        return myList;
+    }
+
+    @Override
+    public Book findById(String id) throws ItemNotFoundExeption {
```

Comment:
if(myCollection.get(id) == null) {
    throw new ItemNotFoundException("Item not found");
}
return myCollection.get(id);
Be simple))